internal/app/db.go

```diff
@@ -5,14 +5,15 @@ import (
 	"database/sql"
```

Comment:
most of the changes in this file are just moving functions around, that's why the diff is big. but the main changes are:
- introduces a goroutine that triggers `replace` when window interval ticks
- adds a check to see if database is empty

Reply:
sounds good!